Break up installation docs (#3039)
weaveworks/weave-gitops: website/docs/installation/weave-gitops.mdx

---
title: Weave GitOps
sidebar_position: 2
hide_title: true
pagination_next: "getting-started"
---

## Installing Weave GitOps

> #### Install Flux

Reviewer: There is already a section with the exact same name under "Before you begin" so I think what we should do is move the content of this section to the other one.

> Weave GitOps is an extension to Flux and therefore requires that Flux 0.32 or later has already been installed on your Kubernetes cluster. Full documentation is available at: [https://fluxcd.io/docs/installation/](https://fluxcd.io/docs/installation/).

Reviewer: ```suggestion
```
We have a section for installing Flux on this very page so telling people they need to have Flux already installed just to tell them how to install Flux right after is a little strange.

This version of Weave GitOps is tested against the following Flux releases:
* 0.32
* 0.33
* 0.34
* 0.35

> * 0.36

Reviewer: This should go in the "Install Flux" section below.

## Before you begin

We will provide a complete walk-through of getting Flux installed and Weave GitOps configured. However, if you have:
- an existing cluster bootstrapped Flux version >= 0.32.0 🎉
- followed our [installation](./overview.mdx) doc to configure access to the Weave GitOps dashboard then install Weave GitOps 👏

> Then you can skip ahead to [Part 1](#part-1---weave-gitops-overview) 🏃

Reviewer: ```suggestion
Then you can skip ahead to [the Weave GitOps overview](../getting-started.mdx#part-1---weave-gitops-overview) 🏃
```

but note ⚠️ you may need to alter commands where we are committing files to GitHub ⚠️.

To follow along, you will need the following:
- A Kubernetes cluster - such as [Kind](https://kind.sigs.k8s.io/docs/user/quick-start/).
- A [GitHub](https://github.com) account and [personal access token with repo permissions](https://help.github.com/github/authenticating-to-github/creating-a-personal-access-token-for-the-command-line).
- [kubectl](https://kubernetes.io/docs/tasks/tools/#kubectl).

> ### Install Flux

Reviewer: ```suggestion
## Install Flux
```

1. Install the flux CLI

   ```
   brew install fluxcd/tap/flux
   ```

   For other installation methods, see the relevant [Flux documentation](https://fluxcd.io/docs/installation/#install-the-flux-cli).

1. Export your credentials

   ```
   export GITHUB_TOKEN=<your-token>
   export GITHUB_USER=<your-username>
   ```

1. Check your Kubernetes cluster

   ```
   flux check --pre
   ```

   The output is similar to:
   ```
   ► checking prerequisites
   ✔ kubernetes 1.22.2 >=1.20.6
   ✔ prerequisites checks passed
   ```

1. Install Flux onto your cluster with the `flux bootstrap` command

   ```
   flux bootstrap github \
     --owner=$GITHUB_USER \
     --repository=fleet-infra \
     --branch=main \
     --path=./clusters/my-cluster \
     --personal
   ```

Full installation documentation including how to work with other Git providers is available [here](https://fluxcd.io/docs/installation/).

> The bootstrap command above does following:

Reviewer: ```suggestion
The bootstrap command above does the following:
```

> - Creates a git repository fleet-infra on your GitHub account
> - Adds Flux component manifests to the repository
> - Deploys Flux Components to your Kubernetes Cluster
> - Configures Flux components to track the path /clusters/my-cluster/ in the repository

Reviewer: ```suggestion
- Create a git repository fleet-infra on your GitHub account
- Add Flux component manifests to the repository
- Deploy Flux Components to your Kubernetes Cluster
- Configure Flux components to track the path /clusters/my-cluster/ in the repository
```
